sub-packages/bionemo-esm2/tests/bionemo/esm2/model/test_convert.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+
+import pytest
+import torch
+from nemo.lightning import io
+from transformers import AutoModelForMaskedLM
+
+from bionemo.core.data.load import load
+from bionemo.core.utils.dtypes import get_autocast_dtype
+from bionemo.esm2.data.tokenizer import get_tokenizer
+from bionemo.esm2.model.model import ESM2Config
+from bionemo.llm.model.biobert.lightning import biobert_lightning_module
+from bionemo.testing import megatron_parallel_state_utils
+
+
+tokenizer = get_tokenizer()
+
+
+def run_esm2_ckpt_conversion_hf_to_nemo(ckpt_path: Path, model_tag: str):
+    from bionemo.esm2.model.convert import HFESM2Importer  # noqa: F401
+
+    module = biobert_lightning_module(config=ESM2Config())
+    io.import_ckpt(module, f"hf://{model_tag}", ckpt_path / "nemo_checkpoint")
```

Comment:
Just curious, how is the `apply()` method called here?

Reply:
Yeah it's not obvious, you're "registering" the `io.ModelConnector[AutoModelForMaskedLM, BionemoLightningModule]` for that model type, and then when you pass it the `BionemoLightningModule` it knows which connector class to use